paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/intent/IntentConfirmationInterceptor.kt

```diff
@@ -280,6 +283,31 @@ internal class DefaultIntentConfirmationInterceptor @Inject constructor(
         )
     }
 
+    private suspend fun waitForIntentCallback(): CreateIntentCallback? {
+        return retrieveCallback() ?: run {
+            val callback = withTimeoutOrNull(INTENT_CALLBACK_FETCH_TIMEOUT.seconds) {
+                var intentCallback: CreateIntentCallback? = null
+
+                while (intentCallback == null) {
+                    delay(INTENT_CALLBACK_FETCH_INTERVAL)
```

Comment:
This is adding a delay (although small) for the happy path case (ie, the integration is correct, and the callback already exists).